arangod/Aql/ClusterBlocks.cpp

```diff
@@ -1847,63 +1716,61 @@ bool SingleRemoteOperationBlock::getOne(arangodb::aql::AqlItemBlock* aqlres,
   // only copy 1st row of registers inherited from previous frame(s)
   TRI_ASSERT(result.ok());
   VPackSlice outDocument = VPackSlice::noneSlice();
-  if(result.buffer){
+  if (result.buffer) {
     outDocument = result.slice().resolveExternal();
   }
 
   VPackSlice oldDocument = VPackSlice::noneSlice();
   VPackSlice newDocument = VPackSlice::noneSlice();
-  if(outDocument.isObject()) {
-    if(outDocument.hasKey("old")){
+  if (outDocument.isObject()) {
+    if (outDocument.hasKey("old")){
       oldDocument = outDocument.get("old");
     }
-    if(outDocument.hasKey("new")){
+    if (outDocument.hasKey("new")){
       newDocument = outDocument.get("new");
     }
   }
+  
+  TRI_ASSERT(out || OLD || NEW);
 
-  // place documents as in the outi variable slots of the result
-  bool aqlValueSet = false;
-  if(out) {
-    if(!outDocument.isNone()){
+  // place documents as in the out variable slots of the result
+  if (out) {
+    if (!outDocument.isNone()) {
       aqlres->emplaceValue(outputCounter, static_cast<arangodb::aql::RegisterId>(outRegId), outDocument);
     } else {
       aqlres->emplaceValue(outputCounter, static_cast<arangodb::aql::RegisterId>(outRegId), VPackSlice::nullSlice());
     }
-    aqlValueSet = true;
   }
-  if(OLD) {
+
+  if (OLD) {
     TRI_ASSERT(opOptions.returnOld);
-    if(!oldDocument.isNone()){
+    if (!oldDocument.isNone()) {
       aqlres->emplaceValue(outputCounter, static_cast<arangodb::aql::RegisterId>(oldRegId), oldDocument);
     } else {
       aqlres->emplaceValue(outputCounter, static_cast<arangodb::aql::RegisterId>(oldRegId), VPackSlice::nullSlice());
     }
-    aqlValueSet = true;
   }
-  if(NEW) {
+
+  if (NEW) {
     TRI_ASSERT(opOptions.returnNew);
-    if(!newDocument.isNone()){
+    if (!newDocument.isNone()) {
       aqlres->emplaceValue(outputCounter, static_cast<arangodb::aql::RegisterId>(newRegId), newDocument);
     } else {
       aqlres->emplaceValue(outputCounter, static_cast<arangodb::aql::RegisterId>(newRegId), VPackSlice::nullSlice());
     }
-    aqlValueSet = true;
   }
+
   throwIfKilled();  // check if we were aborted
 
   TRI_IF_FAILURE("SingleRemoteOperationBlock::moreDocuments") {
     THROW_ARANGO_EXCEPTION(TRI_ERROR_DEBUG);
   }
-  if(!aqlValueSet) {
-    return false;
```

Comment:
It cannot happen. Because there were some other checks before that exited the method.
That was actually detected by Codescan, not by me. But I verified it, and the tool was correct. :smile: